contracts/JumpBall.cdc

```diff
@@ -0,0 +1,379 @@
+            let receiver = depositCap.borrow() ?? panic("Failed to borrow receiver capability.")
+            receiver.deposit(token: <-self.nfts.remove(key: nftID)!)
+            emit NFTReturned(gameID: self.id, nftID: nftID, owner: ownerAddress)
+        }
+
+        // Award all NFTs to the winner
+        access(contract) fun transferAllToWinner(winner: Address, winnerCap: Capability<&{NonFungibleToken.Collection}>) {
+            let winnerCollection = winnerCap.borrow() ?? panic("Failed to borrow winner capability.")
+            let keys = self.nfts.keys
+            for key in keys {
+                let nft <- self.nfts.remove(key: key) ?? panic("NFT not found.")
+                let previousOwner = self.ownership.remove(key: key)!
+                winnerCollection.deposit(token: <-nft)
+                emit NFTAwarded(gameID: self.id, nftID: key, previousOwner: previousOwner, winner: winner)
+            }
+        }
+    }
+
+    // Admin resource for game management
+    access(all) resource Admin {
+        access(contract) fun determineWinner(gameID: String, stats: {UInt64: UInt64}) {
+            let game = &JumpBall.games[gameID] as &Game?
+                ?? panic("Game does not exist.")
+
+            let creatorTotal = self.calculateTotal(game: game, address: game.creator, stats: stats)
+            let opponentTotal = self.calculateTotal(game: game, address: game.opponent, stats: stats)
+
+            if creatorTotal > opponentTotal {
+                self.awardWinner(game: game, winner: game.creator, winnerCap: game.getCreatorCap())
+            } else if opponentTotal > creatorTotal {
+                let opponentCap = game.getOpponentCap() ?? panic("Opponent capability not found.")
+                self.awardWinner(game: game, winner: game.opponent!, winnerCap: opponentCap)
+            } else {
+                game.returnAllNFTs()
+            }
+        }
+
+        access(contract) fun calculateTotal(game: &Game, address: Address?, stats: {UInt64: UInt64}): UInt64 {
+            if address == nil {
+                return 0
+            }
+
+            var total: UInt64 = 0
+            let nftKeys = game.getNFTKeys()
+
+            for key in nftKeys {
+                let owner = game.getOwnership(key: key) ?? panic("Owner not found.")
+                if owner == address {
+                    total = total + (stats[key] ?? 0)
+                }
+            }
+
+            return total
+        }
+
+        access(contract) fun awardWinner(game: &Game, winner: Address, winnerCap: Capability<&{NonFungibleToken.Collection}>) {
+            emit WinnerDetermined(gameID: game.id, winner: winner)
+            game.transferAllToWinner(winner: winner, winnerCap: winnerCap)
+        }
+    }
+
+    // Factory function to create a new Player resource
+    access(all) fun createPlayer(playerAddress: Address, collectionCap: Capability<&{NonFungibleToken.Collection}>): @Player {
```

Comment:
Any `playerAddress` and `collectionCap` can be passed here not necessarily those of the account calling this method

We might be able to address this:
- By using Player resource UUIDs rather than addresses to tie games to users so that addresses are not needed anymore, or
- By using an entitled capability param here and getting the address from the capability itself, though we should avoid storing the capability if it's not useful (it looks like `collectionCap` is not actually used?). Also the address value would remain the same in case the player resource is transferred to someone else and there may be limitations related to that